pkg/rollout/rollout.go

```diff
@@ -142,26 +172,42 @@ func (r *Rollout) SplitTraffic(svc *run.Service, stable, candidate string) *run.
 		traffic = append(traffic, stableTraffic)
 	}
 	traffic = append(traffic, candidateTraffic)
-
-	// Respect tags manually introduced by the user (e.g. UI/gcloud).
-	customTags := userDefinedTrafficTags(svc)
-	traffic = append(traffic, customTags...)
-
-	// Always assign latest tag to the latest revision.
-	traffic = append(traffic, &run.TrafficTarget{LatestRevision: true, Tag: LatestTag})
+	traffic = append(traffic, inheritRevisionTags(svc)...)
 
 	if !r.promoteToStable {
-		r.log.Infof("will assign %d%% of the traffic to stable revision %s", stablePercent, stable)
-		r.log.Infof("will assign %d%% of the traffic to candidate revision %s", candidateTraffic.Percent, candidate)
+		r.log.Infof("will assign %d%% of the traffic to stable revision", stablePercent)
+		r.log.Infof("will assign %d%% of the traffic to candidate revision", candidateTraffic.Percent)
 	} else {
 		r.log.Infof("will make revision %s stable", candidate)
 	}
 
 	svc.Spec.Traffic = traffic
+	return svc
+}
 
+// PrepareRollback redirects all the traffic to the stable revision.
+func (r *Rollout) PrepareRollback(svc *run.Service, stable, candidate string) *run.Service {
+	traffic := []*run.TrafficTarget{
+		newTrafficTarget(stable, 100, StableTag),
+		newTrafficTarget(candidate, 0, CandidateTag),
+	}
+	traffic = append(traffic, inheritRevisionTags(svc)...)
+
+	svc.Spec.Traffic = traffic
 	return svc
 }
 
+// replaceService updates the service object in Cloud Run.
+func (r *Rollout) replaceService(svc *run.Service) (*run.Service, error) {
```

Comment:
prob should return just `error` you don't use the returned service. (ditto for UpdateService)